staging/src/k8s.io/apimachinery/pkg/util/validation/field/errors.go

```diff
@@ -58,7 +58,12 @@ func (v *Error) ErrorBody() string {
 			if reflectValue := reflect.ValueOf(value); reflectValue.IsNil() {
 				value = "null"
 			} else {
-				value = reflectValue.Elem().Interface()
+				var ok bool
+				// It is possible for the pointer to a type to be a Stringer, but not the type itself.
+				// If the pointer is a Stringer, then use the Stringer implementation of the pointer.
+				if value, ok = reflectValue.Interface().(fmt.Stringer); !ok {
```

Comment:
This change isn't obviously relate the addition of the nil check in metav1.Time.  Is this fixing a separate problem?  Can you craft a test demonstrating the problem and solution I can try before and after?

Reply:
> This change isn't obviously relate the addition of the nil check in metav1.Time. Is this fixing a separate problem? Can you craft a test demonstrating the problem and solution I can try before and after?

I have since found https://github.com/kubernetes/kubernetes/pull/105982#issuecomment-958653511, but the changes noted https://github.com/kubernetes/kubernetes/pull/105982#discussion_r742312742 and to https://github.com/kubernetes/kubernetes/pull/105982/files#r742307044 look unusual to me.  I wouldn't expect to see consumers of things like `ErrorList` to have to change.

Reply:
I think this is best illustrated with an example: https://play.golang.org/p/U0NiGXgNwjZ

In the example, you can see that even though `*Time` implements the `fmt.Stringer`, it will not be recognized as such when formatting the error in the `errorFormattingOld`. With the changes (`errorFormattingNew`), the `*Time` variable is properly recognized as a `fmt.Stringer`.

Reply:
Another approach here could be to add an explicit `Time` case in the switch. This would be more explicit and only affect `v1.Time`s.

Reply:
neither of those options seems correct to me... why are consumers having to react like this?

Reply:
The embedded `time.Time` [implements String() on the struct, not on a pointer receiver](https://cs.opensource.google/go/go/+/refs/tags/go1.17.2:src/time/format.go;l=497)

Can we do the same on our implementation? Would that help the debugger-dereference issue? It seems like a debugger would already have to handle not dereferencing a nil pointer to get to a non-pointer-receiver String impl.